trestlebot/tasks/authored/compdef.py

```diff
@@ -158,6 +169,63 @@ def create_new_default(
         )
         rules_view_builder.write_to_yaml(rule_dir)
 
+    def create_update_cac_compdef(
+        self,
+        comp_description: str,
+        comp_type: str,
+        product: str,
+        cac_content_root: str,
+        working_dir: str,
+    ) -> None:
+        """Create component definition for cac content
+
+        Args:
+            comp_description: Description of the component
+            comp_type: Type of the component
+            product: Product name for the component
+            cac_content_root: ComplianceAsCode repo path
+            working_dir: workplace repo path
+        """
+        # Initial component definition fields
+        component_definition = generate_sample_model(ComponentDefinition)
+        component_definition.metadata.title = f"Component definition for {product}"
+        component_definition.metadata.version = "1.0"
+        component_definition.components = list()
+        oscal_component = generate_sample_model(DefinedComponent)
+        oscal_component.title = get_component_title(product, cac_content_root)
+        oscal_component.type = comp_type
+        oscal_component.description = comp_description
+
+        # Create all of the component properties for rules
+        # This part will be updated in CPLYTM-218
+        """
+        rules: List[RuleInfo] = self.rules_transformer.get_all_rules()
+        all_rule_properties: List[Property] = self.rules_transformer.transform(rules)
+        oscal_component.props = none_if_empty(all_rule_properties)
+        """
+        repo_path = pathlib.Path(working_dir)
+        out_path = repo_path.joinpath(f"{const.MODEL_DIR_COMPDEF}/{product}/")
+        oname = "component-definition.json"
+        ofile = out_path / oname
+        if ofile.exists():
+            logger.info(f"The component for product {product} exists.")
+            with open(ofile, "r", encoding="utf-8") as f:
+                data = json.load(f)
+            for component in data["component-definition"]["components"]:
+                if component.get("title") == get_component_title(
```

Comment:
Could it be compared with `oscal_component.title` to avoid calling this function twice?

Reply:
Good catch. Updated.